crypto/hpke/hpke.c

```diff
@@ -143,112 +138,94 @@ static EVP_PKEY *EVP_PKEY_new_raw_nist_public_key(OSSL_LIB_CTX *libctx,
  * @param ivlen is the length of the iv
  * @param aad is the additional authenticated data
  * @param aadlen is the length of the aad
- * @param cipher is obvious
- * @param cipherlen is the ciphertext length
- * @param plain is an output
- * @param plainlen input/output, better be big enough on input, exact on output
- * @return 1 for good otherwise bad
+ * @param ct is the ciphertext buffer
+ * @param ctlen is the ciphertext length
+ * @param pt is a caller-allocated output buffer
+ * @param ptlen input/output, better be big enough on input, exact on output
+ * @return 1 on success, 0 otherwise
  */
 static int hpke_aead_dec(OSSL_LIB_CTX *libctx, const char *propq,
                          OSSL_HPKE_SUITE suite,
                          const unsigned char *key, size_t keylen,
                          const unsigned char *iv, size_t ivlen,
                          const unsigned char *aad, size_t aadlen,
-                         const unsigned char *cipher, size_t cipherlen,
-                         unsigned char *plain, size_t *plainlen)
+                         const unsigned char *ct, size_t ctlen,
+                         unsigned char *pt, size_t *ptlen)
 {
-    int erv = 1;
+    int erv = 0;
     EVP_CIPHER_CTX *ctx = NULL;
     int len = 0;
-    size_t plaintextlen = 0;
-    unsigned char *plaintext = NULL;
     size_t taglen;
     EVP_CIPHER *enc = NULL;
     const OSSL_HPKE_AEAD_INFO *aead_info = NULL;
 
+    if (pt == NULL || ptlen == NULL) {
+        ERR_raise(ERR_LIB_CRYPTO, ERR_R_INTERNAL_ERROR);
+        goto err;
+    }
     aead_info = ossl_HPKE_AEAD_INFO_find_id(suite.aead_id);
     if (aead_info == NULL) {
-        erv = 0;
         ERR_raise(ERR_LIB_CRYPTO, ERR_R_INTERNAL_ERROR);
         goto err;
     }
     taglen = aead_info->taglen;
-    plaintext = OPENSSL_malloc(cipherlen);
-    if (plaintext == NULL) {
-        erv = 0;
+    if (*ptlen < (ctlen - taglen)) {
```

Comment:
Can you test what happens when ctlen < taglen.. These are size_t values..

Reply:
ah good yes, changed to ``if ((*ptlen + taglen) < ctlen)`` with similar change in _open and added test calls to test with too-short output buffers